### PR TITLE
tests(channels): Improve channels testing

### PR DESCRIPTION
**Pull request by provinzkraut**
Improve the tests for channels by covering a few untested edge cases

### CODE REVIEW

File: docs/usage/channels.rst [hunk 1]
```diff
@@ -399,7 +399,7 @@ implemented are:
     A basic in-memory backend, mostly useful for testing and local development, but
     still fully capable. Since it stores all data in-process, it can achieve the highest
     performance of all the backends, but at the same time is not suitable for
-    applications running on multiple processes.
+    applications running on multiple processes
```
Comment (cofin) [inline docs/usage/channels.rst:402]:
```suggestion
    applications running on multiple processes.
```

**Review by cofin — APPROVED** [review]
LGTM

**Review by JacobCoffee — APPROVED** [review]

**Comment by JacobCoffee** [comment]
I just saw that as I review :(

**Review by cofin — APPROVED** [review]
